skrub/_fuzzy_join.py

```diff
@@ -506,9 +506,11 @@ def fuzzy_join(
     if drop_unmatched:
         df_joined.drop(columns=["fj_idx"], inplace=True)
     else:
-        idx = df_joined.index[df_joined["fj_nan"] == 1]
-        if len(idx) != 0:
-            df_joined.iloc[idx, df_joined.columns.get_loc("fj_idx") :] = np.NaN
+        mask_na = df_joined["fj_nan"] == 1
+        if mask_na.sum() > 0:
```

Comment:
```suggestion
        if mask_na.any():
```
I would rather use "any", rather than "sum". It's more readable, and it's also faster (though we are in fast operations):
```
In [2]: a = np.zeros(10_000, dtype=bool)

In [3]: a[100] = 1

In [4]: %timeit np.sum(a) > 0
7.13 µs ± 58 ns per loop (mean ± std. dev. of 7 runs, 100,000 loops each)

In [5]: %timeit np.any(a)
2.7 µs ± 30.6 ns per loop (mean ± std. dev. of 7 runs, 100,000 loops each)

In [6]: %timeit a.any()
1.03 µs ± 7.16 ns per loop (mean ± std. dev. of 7 runs, 1,000,000 loops each)

In [7]: %timeit a.sum() > 0
5.8 µs ± 376 ns per loop (mean ± std. dev. of 7 runs, 100,000 loops each)
```